### PR TITLE
Updated QA toolchain

### PR DESCRIPTION
**Pull request by weierophinney**
- Added PHPUnit and php-cs-fixer as dev dependencies.
- Setup PSR-0 dev autoloading for tests.
- Added php-cs-fixer configuration.
- Updated Travis-CI configuration to:
 - Add testing for 5.5, 5.6, 7, and hhvm.
 - Use the phpunit and php-cs-fixer installed by composer.
 - Remove xdebug support (as we're not generating coverage).
 - Only run CS checks on 5.6.

Additionally, once the above were complete, ran php-cs-fixer without the `--dry-run` flag to fix CS issues.

### CODE REVIEW

**Comment by Maks3w** [comment]
append || return 0 to xdebug 
use 1.7.* as constrait of cs tool

**Comment by weierophinney** [comment]
@Maks3w —

> append || return 0 to xdebug

Already did; see the most recent commit.

> use 1.7.* as constrait of cs tool

`~1.7.0` is functionally equivalent anyways (will install anything between 1.7.0 and 1.8.0, not including 1.8.0), and wildcards are frowned upon by the Composer team.

**Comment by Maks3w** [comment]
I don't like the DISABLE_XDEBUG config option.
Alone does not have sense.
Require to manual change when xdebug become available on PHP7 or HHVM

**Comment by weierophinney** [comment]
@Maks3w will the `|| return 0` correct for that? If so, send a PR removing the variable and the conditional.

File: composer.json [hunk 1]
```diff
@@ -17,6 +17,11 @@
             "ZendService\\Amazon\\": "library/"
         }
     },
+    "autoload-dev": {
+        "psr-0": {
+            "ZendServiceTest\\Amazon\\": "tests/ZendServiceTest/Amazon/"
```
Comment (Maks3w) [inline composer.json]: Is this PSR-0 or PSR-4?

the given path is for PSR-4

Comment (weierophinney) [inline composer.json]: It's PSR-0; in fact, autoloading failed when I initially tried that as PSR-4.

File: tests/ZendService/Amazon/SimpleDb/OnlineTest.php [hunk 2]
```diff
@@ -364,23 +364,23 @@ public function testListDomains()
             // Amazon returns an empty page as the last page :/
             $isLast = $page->isLast();
             if (!$isLast) {
-              // The old isLast() assertTrue failed in full suite runs. Token often
+                // The old isLast() assertTrue failed in full suite runs. Token often
               // decodes to 'TestsZendServiceAmazonSimpleDbDomain_testPutAttributes'
```
Comment (Maks3w) [inline tests/ZendService/Amazon/SimpleDb/OnlineTest.php]: Seems CS tool didn't indent this as should be

Comment (weierophinney) [inline tests/ZendService/Amazon/SimpleDb/OnlineTest.php]: Looks like it properly indented the line below the conditional, but the next 5 lines were not indented correctly.

File: composer.json (hunk 1 above)
Comment (Maks3w) [inline composer.json]: I'm sure PSR-0 and PSR-4 are mixed in this syntax. PSR-0 syntax should be simmetric to "require"

Comment (weierophinney) [inline composer.json]: It's correct. I've tried each, and tried different sets of paths. The `autoload` entry is *verbatim* from the previous, and looks exactly like this as well.

We cannot do `"ZendServiceTest\\": "tests/ZendServiceTest/"`, as we want to be more specific than that; thus the `Amazon` segments. I originally assumed PSR-4 would work as well, but had autoloading issues when I changed the autoloading type and ran `dump-autoload`. Reverting and re-dumping the autoloader caused it to work.

**Pull request merged**